nasa/harmony-gdal-adapter
# Implement context manager for osgeo.gdal.Open.

## Description

This PR primarily implements a context manager for `osgeo.gdal.Open`, so that files can be opened (and closed) with standard Python syntax:

```
from gdal_subsetter.utilities import OpenGDAL

with OpenGDAL('file.tiff') as gdal_dataset:
    gdal_metadata = gdal_dataset.GetMetadata()
```

## Jira Issue ID

N/A

## Local Test Steps

* Pull this branch locally.
* Build the HGA service image (`bin/build-image`)
* Restart your local Harmony instance (`bin/restart-services`)
* Run the [HGA regression test notebook](https://github.com/nasa/harmony-regression-tests/blob/main/test/hga/HGA_Regression.ipynb). All tests should still pass.

## PR Acceptance Checklist
* ~~Jira ticket acceptance criteria met.~~
* ~~Documentation updated (if needed).~~
